src/arrayMethodSort.js

```diff
@@ -4,9 +4,41 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction = defoultSortFunction) {
+    for (let i = 0; i < this.length - 1; i++) {
+      let current = this[i];
+
+      for (let j = i + 1; j < this.length; j++) {
+        const next = this[j];
+
+        if (compareFunction(current, next) > 0) {
+          current = next;
+        }
+      }
+
+      const temp = this[i];
+
+      this[this.indexOf(current)] = temp;
+      this[i] = current;
```

Comment:
Using 'this.indexOf(current)' is inefficient because it searches through the array to find the index, which increases the time complexity. Since you are already in the inner loop, you should use 'j' as the index to swap with 'i'.